### PR TITLE
Fix #1367 - Move to Boost LICENSE

### LINKED ISSUE #1367: [IMPORTANT] [LICENCE] Move from MIT to BSL licence
Recent discussions with some people interested in using EVE led us to think about moving from MIT to Boost Software Licence.
I'm rather OK with it but we formally need the go from contributors to migrate there. 

So would:
 - [x] @jfalcou myself
 - [x] @jtlap
 - [x] @DenisYaroshevskiy
 - [x] @aguinet
 - [x] @justend29
 - [x] @the-moisrex
 - [ ] @Simran-B
 - [x] @HadrienG2
 - [x] @toughengineer
 - [x] @pauljurczak
 - [x] @JPenuchot
 - [x] @orao

be keen to answer this issue with a YAY or NAY so we can move forward if everybody is OK with the move.
Thanks, everyone

### PR DESCRIPTION
**Pull request by jfalcou**

### CODE REVIEW

File: tools/find_deps
Comment (DenisYaroshevskiy) [inline tools/find_deps:4]: Hey - this looks lie a typo in a couple of places

File: include/eve/arch/tags.hpp
Comment (DenisYaroshevskiy) [inline include/eve/arch/tags.hpp:5]: hey

Comment (jfalcou) [reply include/eve/arch/tags.hpp]: Fixed all of them

**Review by DenisYaroshevskiy — APPROVED** [review]

**Comment by jfalcou** [comment]
Need to wait for doc PR from @jtlap before merged.

**Pull request merged**
